src/extension.ts

```diff
@@ -56,10 +57,16 @@ function getSearchUrl(query: string) {
 	let defaultProvider = config.get<string>("defaultProvider");
 	let providerID = query.split(' ', 1)[0];
 	
+	// Represents the default defaultProvider, to tell difference between default and user defined default.
+	let defaultProviderDefault = "b"
+	
 	// Backwards compatibility with old config format
 	let oldSearchProvider = config.get<string>("searchprovider");
 	if (oldSearchProvider != null) {
 		defaultProvider = oldSearchProvider;
+        vscode.window.showWarningMessage("codebing.searchprovider is depricated!", "Open global config").then(() => {
```

Comment:
We should also provide the option to open the workspace settings if a folder is opened:
```
		// Setup warning
		interface CmdItem extends vscode.MessageItem { cmd: string };
		let openGlobalSettings:CmdItem = { title: "Open global settings", cmd: "workbench.action.openGlobalSettings"};
		let openWorkspaceSettings:CmdItem = { title: "Open workspace settings", cmd: "workbench.action.openWorkspaceSettings"};
		let warning = "codebing.searchprovider is deprecated!";
		
		// Only show "Open workspace settings" if a folder is open
		(vscode.workspace.rootPath == undefined
				? vscode.window.showWarningMessage(warning, openGlobalSettings)
				: vscode.window.showWarningMessage(warning, openGlobalSettings, openWorkspaceSettings))
			.then((c) => {
				if (c)
					vscode.commands.executeCommand(c.cmd);
			});
```
... yes I know it is a little bit ugly but there has to be a function call for each situation because of this god damn vararg function. And it can not handle undefined or null parameters.

Hmm maybe put it into its own function :)